File: H_Simulator_FVM_1D/03_H_Simulator_FVM_1D_Compressible_Homogeneous/MATLAB/src/Add_BC.m
function [BC,dBC] = Add_BC(dRhodP,Gen,Rho,State,Trans)

%Predefine memory
BC = zeros(Gen.Nx,1);                                           %[N,1]
dBC = zeros(Gen.Nx,1);                                          %[N,1]

%Find "source" terms from boundary conditions
BC(1) = (Trans.x/2) * Rho(1) * (State.P(1)  - Gen.PL);          %[N,1]
BC(end) = (Trans.x/2) * Rho(end) * (State.P(end)- Gen.PR);      %[N,1]


dBC(1) = (Trans.x/2) * (Rho(1) + dRhodP(1)*(State.P(1) - Gen.PL));
dBC(end) = (Trans.x/2) * (Rho(end) + dRhodP(end)*(State.P(end) - Gen.PR));

end

File: H_Simulator_FVM_1D/04_H_Simulator_FVM_1D_Compressible_Heterogeneous/Python/src/add_bc.py
import numpy as np

def add_bc(dRhodP, Gen, Rho, State, trans):

    #Predefine memory
    BC = np.zeros(Gen["Nx"])
    dBC = np.zeros(Gen["Nx"])

    #Find "source" terms from boundary conditions
    BC[0] = (trans["x"][0] / 2) * (Rho[0] * (State["P"][0] - Gen["PL"]))
    BC[-1] = (trans["x"][-1] / 2) * (Rho[-1] * (State["P"][-1] - Gen["PR"]))

    dBC[0] = (trans["x"][0] / 2) * (Rho[0] + dRhodP[0] * (State["P"][0] - Gen["PL"]))
    dBC[-1] = (trans["x"][-1] / 2) * (Rho[-1] + dRhodP[-1] * (State["P"][-1] - Gen["PR"]))

    return BC, dBC